### PR TITLE
[PUB-1609] Path for new business trialists' empty dashboard

### PR DESCRIPTION
**Pull request by chayjay**
## Description

Adding new path for business trialists to hit empty dashboard when they have no social accounts connected.

## Context & Notes
This change is required to ensure Business Trialists have a smooth on-boarding experience.
JIRA: https://buffer.atlassian.net/browse/PUB-1609?atlOrigin=eyJpIjoiMDdhMjhjZGRmMTJmNDI1MmJmYzFhZjEwNDI5NDdjMTAiLCJwIjoiaiJ9

## Screenshots (if appropriate)
http://hi.buffer.com/6e1662861041
## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [x] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [x] I have commented my code, particularly in hard-to-understand areas.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/profile-sidebar/middleware.js [hunk 1]
```diff
@@ -70,6 +70,9 @@ export default ({ dispatch, getState }) => next => (action) => {
         path,
       });
       const profiles = getState().profileSidebar.profiles;
+      const { productFeatures: { planName } } = getState();
+      const onTrial = getState().appSidebar.user.trial.onTrial;
+      const isOnBusinessTrial = planName === 'business' && onTrial;
```
Comment (anafilipadealmeida) [inline packages/profile-sidebar/middleware.js]: Would it make sense to move the isOnBusinessTrial logic to the userParser and use just the result coming from the user request? Something like `isOnBusinessTrial: isOnBusinessPlan(userData.trial_plan)` on userParser.js

Comment (chayjay) [reply packages/profile-sidebar/middleware.js]: @anafilipadealmeida, I think that makes total sense! It'll allow me to clean up the logic in other places too. I'll work on it right away ☺

**Comment by chayjay** [comment]
Hey @anafilipadealmeida, would love to circle back to this with you tomorrow - I think the logic here will need to be changed to address a race condition I keep running into. I can only get the new path to render in like 1 in 10 cases. I believe this is because isOnBusinessTrial is being evaluated as undefined most of the time due to the user data not being available at the time of evaluation.

**Comment by anafilipadealmeida** [comment]
Hey @chayjay, great catch! 
You probably don't have any problems in the modals/middleware, as you only need the prop once the user request was successful, after `user_${dataFetchActionTypes.FETCH_SUCCESS}`
However, in the profileSidebar, the first user request was still not made. I assume it is because the first user request is made on the appSidebar, an external package that is rendered in the Sidebar component. Even though the user request is made on APP_INIT, it seems to be the last of the initial requests to be made.

**Comment by anafilipadealmeida** [comment]
@hamstu, summing up: we need to have access to the user data in the profile-sidebar middleware, to know if the user is on a Business Trial. However sometimes the profiles request was happening before the user request. Our solution: instead of making the profiles request on `APP_INIT`, to do it after `user_${dataFetchActionTypes.FETCH_SUCCESS}`. It seems to be working, and we are assuming there's not a situation where the user request is not done on APP_INIT and we would still need the profiles request to happen. However we'd love to hear your thoughts!

**Comment by MayaUribe** [comment]
@anafilipadealmeida @chayjay that solution you came up with makes a lot of sense to me, I think that as long as we are making the `fetch->user` request on `APP_INIT`, this is looking good to me 😄

**Comment by chayjay** [comment]
@hamstu thank you so much!! I made changes based on your recommendations. How does it look now?

**Comment by hamstu** [comment]
Looks great @chayjay! 🙌

**Comment by MayaUribe** [comment]
Hey there @chayjay @anafilipadealmeida I deployed my changes with the Tooltip and merged them into this PR 😄, so I believe this is our last step to finish up this task 🚀

**Pull request merged**